### DISCUSSION TITLE
Add Remote Op code for set_time ( int 'epoch & Unix timestamp')  to do easy manual time sync

### DISCUSSION THREAD

**Discussion by hansju-11** [cyberman54/ESP32-Paxcounter#760] [Ideas]
I would like to add a remote command to sync the paxcounter by hand. So if you do not need a accurate time but a "kind of time" it may be a good option to send the time as an int epoch_timestamp and let the device run internaly the clock. At least you have a near timestamp on the sd card.
 I'm using a TTGO T3 V1.6 without GPS. I don't want to implement a MQTT or application timeserver. We can use the next free opcode 0x88 as an example.

Are there a need for that?

**Comment by cyberman54** [comment]
@hansju-11 Today this has been implemented in development branch:
https://github.com/cyberman54/ESP32-Paxcounter/commit/b5314f1288094c45a63c22e6f337faa6fab458a6

Could you please test, and open issue, if any problems occur? Thanks.

**Reply by hansju-11** [reply]
see #773 .
The code is not working because of the new integration of libpax helper. I was not able to the this new opcode.

**Reply by cyberman54** [reply]
Unfortunately libpax-only branch was accidentially merged with development branch, what broke the development build :-(
The development code currently compiles, if you disable the display in board's hal file:

`//#define HAS_DISPLAY 1`

**Reply by hansju-11** [reply]
see #774
 I think there is a problem in timekeeper. 
Second problem. After the first send the system calls  _deepsleep():

...
[I][rcommand.cpp:344] get_time(): Remote command: get time
[I][lorawan.cpp:226] lora_send(): 5 byte(s) sent to LORA
[I][lorawan.cpp:576] SaveLMICToRTC(): LMIC state saved
[I][reset.cpp:182] enter_deepsleep(): Going to sleep, good bye. ..

and never wakes up again. 

For testing I just remove the code in reset.cpp -> enter_deepsleep() {}

**Reply by cyberman54** [reply]
#773 and #774 are now fixed in development branch.

**Reply by hansju-11** [reply]
it looks like you removew the opcode 88 in development. I will check this at the weekend deeper.

**Reply by cyberman54** [reply]
Should be in there, and i tested it. Time was set.

![image](https://user-images.githubusercontent.com/10289836/113310054-14a04000-9308-11eb-8a9c-fed8f8799925.png)

**Comment by hansju-11** [comment]
works greate. thank you @cyberman54 
We can close this discussion.
